### PR TITLE
Experiment reload

### PR DESCRIPTION
**Pull request by snjqi188**
Happy to say that this code seems to work. 

We clicked around different tabs and forum topics WITHOUT ever refreshing. After 30 minutes of fiddling around, the ad still appears. 

Please try it out and let us know how it goes.

However, there are 2 issues that we've identified:

**1) Despite using PageTracker, the ads don't change between tabs.**

localhost:3000/unread

screen shot 2015-08-12 at 12 04 44 pm

localhost:3000/new

screen shot 2015-08-12 at 12 04 48 pm

Could this be an issue with PageTracker, or is it because it's still connected to the same plugin outlet?
We hope to get something similar to [this site](ozim.co.li) , where they've managed to get the ad to reload between tabs. Click `/unread` and `/new`, you can see that the ad changes.

**2) The number of warnings increase everytime we load an ad.**

The first image shows 24 Warnings, and after navigating around different pages, for another 24 times, we have 42 Warnings. Should this be a concern?

screen shot 2015-08-12 at 11 56 40 am
screen shot 2015-08-12 at 12 03 19 pm

### CODE REVIEW

File: assets/javascripts/discourse/components/google-adsense.js.es6 [hunk 1]
```diff
@@ -7,30 +7,57 @@ var ad_height = '';
...
+
+    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true; ga.id="adsense_loader";
+    ga.src = '//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js';
+    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
+
+    __reload();
+})
+
+function __reload () {
+    if('' != Discourse.SiteSettings.adsense_publisher_code) {
```
Comment (adelsmee) [inline assets/javascripts/discourse/components/google-adsense.js.es6]: You don't need the != here - can just do Discourse.SiteSettings.adsense_publisher_code

**Comment by adelsmee** [comment]
@snjqi188 @ladydanger I don't think the number of warnings is an issue, but you should follow the ember link and check out why it is trying to warn you in the first place. If you are creating a security vulnerability that is never a good thing, and they are often easily fixed...

**Comment by eviltrout** [comment]
I tried this out locally and confirmed that `PageTracker` is changing pages correctly. There are a couple of warnings you should fix:

1. you should `import PageTracker from 'discourse/lib/page-tracker' at the top of your file, then you can simply use `PageTracker.current().on(..)` instead of `Discourse.PageTracker`

2. The style bindings are explained here, but basically you have to make a html safe string: http://emberjs.com/deprecations/v1.x/#toc_binding-style-attributes

---

You can confirm PageTracker is working by adding a `console.log` inside your `change` event. You'll see it's called every time a "page" changes. It's hard to tell without digging deeper what the difference is between the other plugin and your component approach is, but there must be something! You might want to compare the output of the other plugin versus yours with the component. You can use chrome's inspector to look through the HTML and see what differences there are. Something must be different!

**Comment by snjqi188** [comment]
@adelsmee  Thanks for the cleaning up code tip! :-)

@eviltrout Thanks for getting back to us! 
1. The PageTracker import didn't work for us. We got an error saying that pagetracker module couldn't be found. Here it is:




2. We are trying to return secure HTML strings. But *Ember.Handlebars.helper* and *return new Handlebars.SafeString* caused `content unsupported` errors. I had this problem in week 2 or 3 and it caused alot of agony. As such, we're getting one of the coaches to look at it.

**Comment by snjqi188** [comment]
Update: Upon Jo's advice on Slack, we've managed to get the container to show after removing the extra function and just return the safestring

however, the ad and relevant iframe code does not show.

```
Ember.Handlebars.helper('google-adsense', function(ad_code, publisher_id){
  return new Handlebars.SafeString('<div style="width:300px; height:250px;"><script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script><ins class="adsbygoogle"style="display:inline-block;width:300px;height:250px;"data-ad-client="ca-pub-' + publisher_id + '"data-ad-slot="' + ad_code + '"></ins><script>(adsbygoogle = window.adsbygoogle || []).push({});</script></div>');
})
```

empty


will be looking at this over the weekend and hopefully some coaches to look at it too

**Comment by ladydanger** [comment]
Hi everyone,
Here's updates.
(1) PageTracker now works!  Thanks heaps.
(2) Variables thing has been sold as Sarah (amazing!) <3 refactored the code.

**Comment by adelsmee** [comment]
The code refactor fixes the problem but it does make the code less reusable. However I can't think of how that is going to necessarily be a problem at this stage, so nice work!

**Pull request closed without merging**